iam.tf

```diff
@@ -42,7 +44,7 @@ resource "aws_iam_role" "apiary_task_readonly" {
        "Sid": "",
        "Effect": "Allow",
        "Principal": {
-         "Service": "ecs-tasks.amazonaws.com"
+         "Service": [ "ecs-tasks.amazonaws.com", "ec2.amazonaws.com" ]
```

Comment:
yes, ideally we should change ecs-task-readwrite to hms-readwrite, this will cause some resources to be recreated when we update existing deployment, if others are ok with the change, will update accordingly.

Reply:
@pradeepbhadani @mroark1m and comment on this?

Reply:
I don't have an issue with it. I wonder if it may fail to deploy cleanly on existing deployments. Because even once the task is updated, the old task will still exist using the old role until ECS gets finished with it's deployment cycle, and so if it has to do create/destroy to update the name, it may balk at doing the destroy. I think that might just mean we have to run it twice? Not sure...

Reply:
terraform was able to make change without issue after adding following lifecycle block to IAM role, otherwise everything seems to work fine.
+  lifecycle {
+    create_before_destroy = true
+  }

Reply:
I will update roles accordingly.

Reply:
I suggest to keep the IAM roles separate as it is a short term limited fix.

Reply:
i thought of the same, but just adds duplicate code, we need to attach same iam policies to both roles, with current setup, we just need to allow both services(ecs&ec2) assume the policy.

Reply:
it adds duplication but if we are going to remove this workaround later than it could help.
Also, Please do not rename the existing task role name as this may break the access to our external buckets (we use explicit Deny and allow to this IAM role).

Reply:
can you provide details? what is the issue if we change the role name? we are controlling external buckets access from the role itself.